packages/venia-concept/src/components/Checkout/wrapper.js

```diff
@@ -0,0 +1,61 @@
+import { Component, createElement } from 'react';
+import { connect } from 'react-redux';
+import { func, shape, string } from 'prop-types';
+
+import timeout from 'src/util/timeout';
+import CheckoutFlow from './flow';
+
+const delay = 1000;
+
+const resetCheckoutAction = () => async dispatch => {
+    dispatch({ type: 'TOGGLE_DRAWER', payload: null });
+    await timeout(192); // drawer transition duration
+    dispatch({ type: 'RESET_CHECKOUT' });
+};
+
+const requestOrderAction = () => async dispatch => {
+    dispatch({ type: 'REQUEST_ORDER' });
+    await timeout(delay); // TODO: replace with api call
+    dispatch({ type: 'RECEIVE_ORDER' });
+};
+
+const submitOrderAction = () => async dispatch => {
+    dispatch({ type: 'SUBMIT_ORDER' });
+    await timeout(delay); // TODO: replace with api call
+    dispatch({ type: 'ACCEPT_ORDER' });
+};
+
+class CheckoutWrapper extends Component {
+    static propTypes = {
+        checkout: shape({
+            status: string
+        }),
+        submitOrder: func
+    };
+
+    render() {
+        const {
+            checkout,
+            resetCheckout,
+            requestOrder,
+            submitOrder
+        } = this.props;
+        const { status } = checkout;
+        const flowProps = { resetCheckout, requestOrder, status, submitOrder };
+
+        return <CheckoutFlow {...flowProps} />;
+    }
+}
+
+const mapStateToProps = ({ checkout }) => ({ checkout });
+
+const mapDispatchToProps = dispatch => ({
+    resetCheckout: () => dispatch(resetCheckoutAction()),
+    requestOrder: () => dispatch(requestOrderAction()),
+    submitOrder: () => dispatch(submitOrderAction())
+});
```

Comment:
Is there an advantage to the dispatch callback, rather than the object form:

```js
const mapDispatchToProps = {
    resetCheckout: resetCheckoutAction,
    requestOrder: requestOrderAction,
    submitOrder: submitOrderAction
};
```
I ask not to be nitpicky, but because I've used the object literal pattern in other PRs and I want to know if you're aware of some drawback to it.

Reply:
No, there's not. I can probably simplify it to match your suggestion. 👍

Reply:
So there is a reason after all.

When a child component calls the function `submitOrder`, `submitOrder` needs to fire an async action (i.e., dispatch an async function). If we don't use `mapDispatchToProps` in this way, then calling `submitOrder` will not be able to fire an action, because `dispatch` will be undefined.

I think this process is correct.